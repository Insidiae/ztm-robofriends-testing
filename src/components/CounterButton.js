import React, { useState } from "react";

const CounterButton = React.memo((props) => {
  const [count, setCount] = useState(0);
  return (
    <button
      id="counter"
      color={props.color}
      onClick={() => setCount(count + 1)}
    >
      Count: {count}
    </button>
  );
});

export default CounterButton;
