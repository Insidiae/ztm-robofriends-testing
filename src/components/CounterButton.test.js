import React from "react";
import { render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import CounterButton from "./CounterButton";

describe("<CounterButton />", () => {
  it("renders properly", () => {
    const mockColor = "red";
    const { container } = render(<CounterButton color={mockColor} />);

    expect.assertions(1);
    expect(container).toMatchSnapshot();
  });

  it("correctly increments the counter", () => {
    const mockColor = "red";
    const { container } = render(<CounterButton color={mockColor} />);

    const btn = container.querySelector("#counter");
    userEvent.click(btn);
    userEvent.click(btn);
    expect(btn).toHaveTextContent("2");
  });
});
