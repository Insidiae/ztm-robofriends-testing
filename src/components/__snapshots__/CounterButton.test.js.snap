// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CounterButton /> renders properly 1`] = `
<div>
  <button
    color="red"
    id="counter"
  >
    Count: 
    0
  </button>
</div>
`;
